File: devlibrary/src/main/java/com/example/devlibrary/utils/ImageLoader.java
package com.example.devlibrary.utils;


import android.graphics.Color;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions;
import com.bumptech.glide.request.RequestOptions;
import com.bumptech.glide.request.transition.DrawableCrossFadeFactory;
import com.example.devlibrary.R;
import com.example.devlibrary.app.App;
import com.example.devlibrary.transform.GlideCircleBorderTransform;
import com.example.devlibrary.transform.GlideRoundedCornersTransform;

/**
 * Created by Administrator on 2017/2/21 0021.
 */

public class ImageLoader {
    private static final String STYLE_CIRCLE = "circle";
    private static final String STYLE_ROUND = "round";
    private static RequestOptions mOptionAvator;

    private static int defImage = 0;
    private static int defHeader = R.mipmap.ic_default_avatar;
    private static String mStyle = STYLE_ROUND;

    public static void loadAvator(ImageView iv, int res) {
        loadAvator(iv, res, defHeader);
    }

    public static void load(ImageView iv, String url) {
        load(iv, url, defImage);
    }

    public static void load(ImageView iv, int res) {
        load(iv, res, defImage);
    }

    public static void loadAvator(ImageView iv, String url) {
        Glide.with(iv.getContext())
                .load(url)
                .apply(getOptionAvator())
                .into(iv);
    }

    public static void loadAvator(ImageView iv, int res, int defaultPicId) {
        Glide.with(iv.getContext())
                .load(res)
                .transition(new DrawableTransitionOptions().crossFade(new DrawableCrossFadeFactory.Builder().setCrossFadeEnabled(true).build()))
                .apply(getOptionAvator().placeholder(defaultPicId))
                .into(iv);
    }

    public static void load(ImageView iv, String url, int defaultPicId) {
        Glide.with(iv.getContext())
                .load(url)
                .apply(new RequestOptions().placeholder(defaultPicId).error(defaultPicId))
                .into(iv);
    }

    public static void load(ImageView iv, int res, int defaultPicId) {
        iv.setTag(null);//需要清空tag，否则报错
        Glide.with(iv.getContext())
                .load(res)
                .apply(new RequestOptions().placeholder(defaultPicId))
                .into(iv);
    }


    public static void load(ImageView iv, String url, RequestOptions requestOptions) {
        Glide.with(iv.getContext())
                .load(url)
                .apply(requestOptions)
                .into(iv);
    }

    public static void load(ImageView iv, int res, RequestOptions requestOptions) {
        Glide.with(iv.getContext())
                .load(res)
                .apply(requestOptions)
                .into(iv);
    }


    public synchronized static RequestOptions getOptionAvator() {
        if (mOptionAvator == null) {
            initHeaderOption();
        }

        return mOptionAvator;
    }

    public synchronized static RequestOptions getOptionAvator(String style) {
        mStyle = style;
        if (mOptionAvator == null) {
            initHeaderOption();
        }

        return mOptionAvator;
    }

    private static void initHeaderOption() {
        if (STYLE_CIRCLE.equals(mStyle)) {
            mOptionAvator = RequestOptions
                    .bitmapTransform(new GlideCircleBorderTransform(10, Color.WHITE))
                    .placeholder(defHeader)
                    .error(defHeader);
        } else {
            mOptionAvator = new RequestOptions()
                    .optionalTransform(new GlideRoundedCornersTransform(App.sContext, 8f, GlideRoundedCornersTransform.CornerType.ALL))
                    .placeholder(defHeader)
                    .error(defHeader);
        }
    }


}

File: baselib/src/main/java/com/wan/baselib/utils/ImageLoader.kt
package com.wan.baselib.utils

import android.graphics.Color
import android.widget.ImageView
import com.bumptech.glide.Glide
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions
import com.bumptech.glide.request.RequestOptions
import com.bumptech.glide.request.transition.DrawableCrossFadeFactory
import com.wan.baselib.R
import com.wan.baselib.ext.appContext
import com.wan.baselib.transform.GlideCircleBorderTransform
import com.wan.baselib.transform.GlideRoundedCornersTransform

/**
 * Created by cy on 13/10/2021.
 */
object ImageLoader {

    private const val STYLE_CIRCLE = "circle"
    private const val STYLE_ROUND = "round"

    private var mOptionAvatar: RequestOptions? = null

    private val defImage = R.mipmap.ic_default_img
    private val defHeader = R.mipmap.ic_default_avatar
    private var mStyle = STYLE_ROUND

    @JvmStatic
    fun load(imageView: ImageView, url: String?) {
        load(imageView, url, defImage)
    }

    @JvmStatic
    fun load(imageView: ImageView, res: Int) {
        load(imageView, res, defImage)
    }

    @JvmStatic
    fun loadAvatar(imageView: ImageView, res: Int) {
        loadAvatar(imageView, res, defHeader)
    }

    @JvmStatic
    fun loadAvatar(imageView: ImageView, url: String?) {
        Glide.with(imageView)
            .load(url)
            .apply(getOptionAvatar())
            .into(imageView)
    }

    @JvmStatic
    fun loadAvatar(imageView: ImageView, res: Int, defaultPicId: Int) {
        Glide.with(imageView)
            .load(res)
            .transition(
                DrawableTransitionOptions().crossFade(
                    DrawableCrossFadeFactory.Builder().setCrossFadeEnabled(true).build()
                )
            )
            .apply(getOptionAvatar().placeholder(defaultPicId))
            .into(imageView)
    }

    @JvmStatic
    fun load(imageView: ImageView, url: String?, defaultPicId: Int) {
        Glide.with(imageView)
            .load(url)
            .apply(RequestOptions().placeholder(defaultPicId).error(defaultPicId))
            .into(imageView)
    }

    @JvmStatic
    fun load(imageView: ImageView, res: Int, defaultPicId: Int) {
        imageView.tag = null //需要清空tag，否则报错
        Glide.with(imageView)
            .load(res)
            .apply(RequestOptions().placeholder(defaultPicId))
            .into(imageView)
    }

    @JvmStatic
    fun load(imageView: ImageView, url: String?, requestOptions: RequestOptions) {
        Glide.with(imageView)
            .load(url)
            .apply(requestOptions)
            .into(imageView)
    }

    @JvmStatic
    fun load(imageView: ImageView, res: Int, requestOptions: RequestOptions) {
        Glide.with(imageView)
            .load(res)
            .apply(requestOptions)
            .into(imageView)
    }

    @Synchronized
    fun getOptionAvatar(): RequestOptions {
        if (mOptionAvatar == null) {
            initHeaderOption()
        }
        return mOptionAvatar!!
    }

    @Synchronized
    fun getOptionAvatar(style: String): RequestOptions {
        mStyle = style
        if (mOptionAvatar == null) {
            initHeaderOption()
        }
        return mOptionAvatar!!
    }

    private fun initHeaderOption() {
        if (STYLE_CIRCLE == mStyle) {
            mOptionAvatar = RequestOptions
                .bitmapTransform(GlideCircleBorderTransform(10f, Color.WHITE))
                .placeholder(defHeader)
                .error(defHeader)
        } else {
            mOptionAvatar = RequestOptions()
                .optionalTransform(
                    GlideRoundedCornersTransform(
                        appContext,
                        8f,
                        GlideRoundedCornersTransform.CornerType.ALL
                    )
                )
                .placeholder(defHeader)
                .error(defHeader)
        }
    }
}